src/multirust-cli/proxy_mode.rs

```diff
@@ -26,6 +26,10 @@ pub fn main() -> Result<()> {
 fn direct_proxy(cfg: &Cfg, arg0: &str) -> Result<()> {
     let cmd = try!(cfg.create_command_for_dir(&try!(utils::current_dir()), arg0));
     let args: Vec<_> = env::args_os().collect();
+
+    if (arg0 == "rustc" || arg0 == "rustc.exe") && cfg.telemetry_enabled() == true {
+        return run_rustc_with_telemetry(cmd, &args, &cfg);
+    }
```

Comment:
It looks to me like all this logic, including `common::run_inner` and `common::run_rustc_with_telemetry` will need to move into `multirust`, perhaps into something like `Cfg::run_command_for_dir`.